src/itensor.jl

```diff
@@ -1277,28 +1266,26 @@ end
 
 -(A::ITensor) = itensor(-tensor(A))
 
-function (A::ITensor{N} + B::ITensor{N}) where {N}
+function (A::ITensor + B::ITensor)
+  if ndims(A) == 0 && ndims(B) > 0
+    return copy(B)
+  end
+  if ndims(B) == 0 && ndims(A) > 0
+    return copy(A)
+  end
+  ndims(A) != ndims(B) && throw(DimensionMismatch("cannot add ITensors with different numbers of indices"))
   C = copy(A)
   C .+= B
   return C
 end
 
-function (A::ITensor{N} - B::ITensor{N}) where {N}
+function (A::ITensor - B::ITensor)
+  ndims(A) != ndims(B) && throw(DimensionMismatch("cannot subtract ITensors with different numbers of indices"))
```

Comment:
We probably want `::ITensor - ::ITensor` to match the behavior of `::ITensor + ::ITensor` in terms of adding to empty order-0 ITensors (or whatever behavior we decide for that as discussed above).

Reply:
Yes

Reply:
I'm confused about where my comments are going (maybe I commented on your branch instead of this PR or something), but this version and the `+` version should also check that in the `ndims(A) == 0` case, you also need that `store(A) isa NDTensors.Empty` in order to get the copying behavior.